src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -129,16 +129,16 @@ def parameter_bytes(self, encoding, format):
         Returns the parameters serialized as bytes.
         """
 
-
-@six.add_metaclass(abc.ABCMeta)
-class DHParametersWithSerialization(DHParameters):
     @abc.abstractmethod
     def parameter_numbers(self):
         """
         Returns a DHParameterNumbers.
         """
 
 
+DHParametersWithSerialization = DHParameters
```

Comment:
We didn't bother with `RSAPublicKeyWithSerialization`, `DSAPublicKeyWithSerialization`, or `EllipticCurvePublicKeyWithSerialization`. If we want to explicitly deprecate the use of these I think we should put that in a separate PR. (For consistency with those other 3 asym key types I've kept the *WithSerialization class docs but we could also remove those in a separate PR)